source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2685 @@
+===================
+Unified Test Format
+===================
+
+:Spec Title: Unified Test Format
+:Spec Version: 1.0.0
+:Author: Jeremy Mikola
+:Advisors: Prashant Mital, Isabel Atkinson, Thomas Reggi
+:Status: Draft
+:Type: Standards
+:Minimum Server Version: N/A
+:Last Modified: 2020-09-15
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+This project defines a unified schema for YAML and JSON specification tests,
+which run operations against a MongoDB deployment. By conforming various spec
+tests to a single schema, drivers can implement a single test runner to execute
+acceptance tests for multiple specifications, thereby reducing maintenance of
+existing specs and implementation time for new specifications.
+
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`__.
+
+
+Goals
+=====
+
+This test format can be used to define tests for the following specifications:
+
+- `Change Streams <../change-streams/change-streams.rst>`__
+- `Command Monitoring <../command-monitoring/command-monitoring.rst>`__
+- `CRUD <../crud/crud.rst>`__
+- `GridFS <../gridfs/gridfs-spec.rst>`__
+- `Retryable Reads <../retryable-reads/retryable-reads.rst>`__
+- `Retryable Writes <../retryable-writes/retryable-writes.rst>`__
+- `Sessions <../sessions/driver-sessions.rst>`__
+- `Transactions <../transactions/transactions.rst>`__
+- `Convenient API for Transactions <../transactions-convenient-api/transactions-convenient-api.rst>`__
+
+This is not an exhaustive list. Specifications that are known to not be
+supported by this format may be discussed under `Future Work`_.
+
+
+Specification
+=============
+
+
+Terms
+-----
+
+Entity
+  Any object or value that is indexed by a unique name and stored in the
+  `Entity Map`_. This will typically be a driver object (e.g. client, session)
+  defined in `createEntities`_ but may also be a
+  `saved operation result <operation_saveResultAsEntity_>`_. A exhaustive list
+  of supported types is presented in `Supported Entity Types`_. Entities are
+  referenced by name throughout the test file (e.g. `Entity Test Operations`_).
+
+Internal MongoClient
+  A MongoClient created specifically for use with internal test operations, such
+  as inserting collection data before a test, configuring fail points during a
```

Comment:
Per https://github.com/mongodb/specifications/commit/8bf4f0bbfb7fefc598ea9cc173953158469c5c22/source/unified-test-format/unified-test-format.rst#diff-840d071b7f5fdbc0931676c81b4ad701L901 failpoints are not configured with the Internal MongoClient, right?

Reply:
Correct. Both `failPoint` and `targetedFailPoint` use a MongoClient under test.